### PR TITLE
Fix premature pausing when reaching end of still-streaming stream

### PR DESCRIPTION
**Pull request by emilk**
### What
Fixes regression caused by https://github.com/rerun-io/rerun/pull/2085/

Closes https://github.com/rerun-io/rerun/pull/2103

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)

PR Build Summary: https://build.rerun.io/pr/2106

### CODE REVIEW

File: crates/re_viewer_context/src/time_control.rs [hunk 1]
```diff
@@ -112,11 +114,17 @@ impl Default for TimeControl {
 
 impl TimeControl {
     /// Update the current time
-    pub fn move_time(&mut self, times_per_timeline: &TimesPerTimeline, stable_dt: f32) {
+    #[must_use]
+    pub fn move_time(
```
Comment (jleibs) [inline crates/re_viewer_context/src/time_control.rs:118]: nit: we've gone pretty far past "moving time" at this point... this is more of `TimeControl::update_state()`

File: crates/re_viewer_context/src/time_control.rs [hunk 2]
```diff
@@ -141,11 +150,17 @@ impl TimeControl {
                     .entry(self.timeline)
                     .or_insert_with(|| TimeState::new(full_range.min));
 
-                if self.looping == Looping::Off && state.time >= full_range.max {
-                    // We've reached the end - stop playing.
+                if self.looping == Looping::Off && full_range.max <= state.time {
+                    // We've reached the end o the data
```
Comment (jleibs) [inline crates/re_viewer_context/src/time_control.rs:154]:
```suggestion
                    // We've reached the end of the data
```

Comment (jleibs) [inline crates/re_viewer_context/src/time_control.rs:154]:
```suggestion
                    // We've reached the end of the data
```

File: crates/re_viewer_context/src/time_control.rs (hunk 1 above)
Comment (emilk) [reply crates/re_viewer_context/src/time_control.rs:118]: The bulk of what it does is move the time though. The only other thing it does is make sure we are on a valid timeline

**Review by jleibs — CHANGES_REQUESTED** [review]
This doesn't work for notebooks yet... probably need to hunt down and figure out why the stream isn't being closed.

File: crates/re_viewer_context/src/time_control.rs (hunk 1 above)
Comment (jleibs) [reply crates/re_viewer_context/src/time_control.rs:118]: And changes the state of the playback controls.

**Comment by emilk** [comment]
> This doesn't work for notebooks yet... probably need to hunt down and figure out why the stream isn't being closed.

Can you help with that? 🙏 I'm heading out now
Feel free to push more stuff to the PR

**Review by jleibs — APPROVED** [review]

**Pull request merged**